src/ansiblelint/utils.py

```diff
@@ -759,6 +760,36 @@ def construct_mapping(
     return data
 
 
+def iter_tasks_in_file(lintable: Lintable, rule_id: str):
+    """Iterate over tasks in file."""
```

Comment:
Be sure that you also document the return values inside the docstring, that is essential when you return more than one value. In that case we yield 4 fields. I checked on typing about the subject as I was not able to find anything using google or SO, got suggested use of https://docs.python.org/3/library/typing.html#typing.NamedTuple as an alternative. 

Still, if that requires many other extra changes we can leave it out for follow-up.

Reply:
Documented. Is that enough or do you think we still need a named tuple?